homeassistant/components/octoprint/sensor.py

```diff
@@ -4,64 +4,57 @@
 
 from pyoctoprintapi import OctoprintJobInfo, OctoprintPrinterInfo
 
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import (
     DEVICE_CLASS_TEMPERATURE,
     DEVICE_CLASS_TIMESTAMP,
     PERCENTAGE,
     TEMP_CELSIUS,
 )
+from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity import Entity
 from homeassistant.helpers.update_coordinator import (
     CoordinatorEntity,
     DataUpdateCoordinator,
 )
 
-from . import CONF_BED, CONF_NUMBER_OF_TOOLS, DOMAIN as COMPONENT_DOMAIN
+from . import DOMAIN as COMPONENT_DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the available OctoPrint sensors."""
-    if discovery_info is None:
-        return
-
-    name = discovery_info["name"]
-    base_url = discovery_info["base_url"]
-    monitored_conditions = discovery_info["sensors"]
-    number_of_tools = discovery_info[CONF_NUMBER_OF_TOOLS]
-    bed = discovery_info[CONF_BED]
-    coordinator: DataUpdateCoordinator = hass.data[COMPONENT_DOMAIN][base_url]
-    tools = []
-    if coordinator.data["printer"]:
-        tools = [tool.name for tool in coordinator.data["printer"].temperatures]
-    else:
-        if number_of_tools > 0:
-            for tool_number in range(number_of_tools):
-                tools.append(f"tool{tool_number!s}")
-        if bed:
-            tools.append("bed")
-
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities
+):
+    """Set up the available OctoPrint binary sensors."""
+    coordinator: DataUpdateCoordinator = hass.data[COMPONENT_DOMAIN][
+        config_entry.entry_id
+    ]["coordinator"]
     entities = []
-    types = ["actual", "target"]
-
-    if "Temperatures" in monitored_conditions and tools:
-        for tool in tools:
+    if coordinator.data["printer"]:
+        printer_info = coordinator.data["printer"]
+        types = ["actual", "target"]
+        for tool in printer_info.temperatures:
             for temp_type in types:
                 entities.append(
-                    OctoPrintTemperatureSensor(coordinator, name, tool, temp_type)
+                    OctoPrintTemperatureSensor(
+                        coordinator,
+                        tool.name,
+                        temp_type,
+                        config_entry.entry_id,
```

Comment:
We're passing the config entry entry_id as device id. Did we decide to do that as the entry unique_id may be missing?

Reply:
Updated to use unique-id

Reply:
This is still using the entry_id.